tests/phpunit/tests/interactivity-api/wpInteractivityAPI.php

```diff
@@ -508,50 +508,130 @@ public function test_process_directives_doesnt_change_html_if_contains_unbalance
+			</header>
 		';
 		$processed_html = $this->interactivity->process_directives( $html );
 		$p              = new WP_HTML_Tag_Processor( $processed_html );
-		$p->next_tag();
+		$p->next_tag( 'svg' );
+		$this->assertNull( $p->get_attribute( 'width' ) );
+		$p->next_tag( 'div' );
+		$this->assertEquals( 'some-id', $p->get_attribute( 'id' ) );
+		$p->next_tag( 'div' );
+		$this->assertEquals( '100', $p->get_attribute( 'id' ) );
+	}
+
+	/**
+	 * Tests that the `process_directives` does not process the HTML
+	 * inside SVG tags.
+	 *
+	 * @ticket 60517
+	 *
+	 * @covers ::process_directives
+	 */
+	public function test_process_directives_does_not_change_inner_html_in_svgs() {
+		$this->interactivity->state(
+			'myPlugin',
+			array(
+				'id' => 'some-id',
+			)
+		);
+		$html           = '
+			<header>
+				<svg height="100">
+					<circle cx="50" cy="50" r="40" stroke="black" stroke-width="3" fill="red" />
+					<div data-wp-bind--id="myPlugin::state.id"></div>
+				</svg>
+			</header>
+		';
+		$processed_html = $this->interactivity->process_directives( $html );
+		$p              = new WP_HTML_Tag_Processor( $processed_html );
+		$p->next_tag( 'div' );
 		$this->assertNull( $p->get_attribute( 'id' ) );
 	}
 
 	/**
-	 * Tests that the `process_directives` returns the same HTML if it finds an
+	 * Tests that the `process_directives` process the HTML outside the
 	 * MathML tag.
 	 *
-	 * @ticket 60356
+	 * @ticket 60517
 	 *
 	 * @covers ::process_directives
 	 */
-	public function test_process_directives_doesnt_change_html_if_contains_math() {
-		$this->interactivity->state( 'myPlugin', array( 'id' => 'some-id' ) );
+	public function test_process_directives_change_html_if_contains_math() {
+		$this->interactivity->state(
+			'myPlugin',
+			array(
+				'id'   => 'some-id',
+				'math' => 'ml-id',
+			)
+		);
 		$html           = '
-			<div data-wp-bind--id="myPlugin::state.id">
-				<math>
+			<header>
+				<math data-wp-bind--id="myPlugin::state.math">
 					<mi>x</mi>
 					<mo>=</mo>
 					<mi>1</mi>
 				</math>
-			</div>
+				<div data-wp-bind--id="myPlugin::state.id"></div>
+			</header>
 		';
 		$processed_html = $this->interactivity->process_directives( $html );
 		$p              = new WP_HTML_Tag_Processor( $processed_html );
-		$p->next_tag();
+		$p->next_tag( 'math' );
+		$this->assertNull( $p->get_attribute( 'id' ) );
+		$p->next_tag( 'div' );
+		$this->assertEquals( 'some-id', $p->get_attribute( 'id' ) );
+	}
+
+	/**
+	 * Tests that the `process_directives` does not process the HTML
+	 * inside MathML tags.
+	 *
+	 * @ticket 60517
+	 *
+	 * @covers ::process_directives
+	 */
+	public function test_process_directives_does_not_change_inner_html_in_math() {
+		$this->interactivity->state(
+			'myPlugin',
+			array(
+				'id' => 'some-id',
+			)
+		);
+		$html           = '
+			<header>
+				<math data-wp-bind--id="myPlugin::state.math">
+				<div data-wp-bind--id="myPlugin::state.id"></div>
```

Comment:
Probably better to use actual MathML, right?

```suggestion
					<mrow data-wp-bind--id="myPlugin::state.id" />
```

Reply:
> The idea was to use an element that we know the server directive processing would process. As far as I know, it should skip SVG tags and its inner accepted elements by default.

Similar here but with MathML